### PR TITLE
Money component

### LINKED ISSUE #172: Money component
Take different currencies and format them in a correct way.

### PR DESCRIPTION
**Pull request by krzkaczor**
image

- [x] money component with formatting
- [x] dummy store to hold decimal values (these should be load from blockchain first
- [x] tests

Closes #172

### CODE REVIEW

File: app/components/shared/Money.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  const decimalPlaces = selectDecimalPlaces(currency);
+  const currencySymbol = selectCurrencySymbol(currency);
+
+  const formattedMoney = doNotSeparateThousands
+    ? formatMoney(value, decimals, decimalPlaces)
+    : formatThousands(formatMoney(value, decimals, decimalPlaces));
+
+  return (
+    <span {...props}>
+      {formattedMoney}{" "}
```
Comment (Moejoe90) [inline app/components/shared/Money.tsx]: Do you need this `{" "}` for a specific reason?

File: app/store.ts [hunk 2]
```diff
@@ -70,7 +72,7 @@ export type IAppState = typeof appStateInstance & {
 export const reducers = combineReducers<IAppState>(allReducers);
 
 interface IAppConnectOptions<S, D, O> {
-  stateToProps?: (state: IAppState) => S;
+  stateToProps?: (state: IAppState, ownProps: O) => S;
```
Comment (Moejoe90) [inline app/store.ts]: Whats the purpose of this?

File: app/components/Demo.tsx [hunk 3]
```diff
@@ -116,5 +117,34 @@ export const Demo: React.SFC = () => (
...
     </Container>
+
+    <Container>
+      <Row>
+        <Col>
+          <Money
+            currency="eth"
+            value={"1234567" + "0".repeat(18)}
+            style={{ fontSize: "25px" }}
+            currencyStyle={{ fontSize: "18px", fontWeight: "bold" }}
```
Comment (lizonp) [inline app/components/Demo.tsx]: Wouldn't it be better to create enum with classNames (ie. small, medum, large)? I am not sure how universal it has to be, though.

File: app/components/shared/Money.tsx (hunk 1 above)
Comment (krzkaczor) [reply app/components/shared/Money.tsx]: yeah - it creates space between value an currency item.

File: app/store.ts (hunk 2 above)
Comment (krzkaczor) [reply app/store.ts]: This allows to access props passed directly to connected component. Actually i think Dave fixed it as well so i am gonna rebase on master

File: app/components/Demo.tsx (hunk 3 above)
Comment (krzkaczor) [reply app/components/Demo.tsx]: Currently, I think designs are wildly different so i made it generic. If we see some similarities among views we can extract them later.

**Pull request merged**